demos/mqtt/mqtt_demo_lightweight/mqtt_demo_lightweight.c

```diff
@@ -0,0 +1,1045 @@
+                                            &packetSize );
+    assert( result == MQTTSuccess );
+    /* Make sure the packet size is less than static buffer size */
+    assert( packetSize < pFixedBuffer->size );
+
+    /* Get next unique packet identifier */
+    unsubscribePacketIdentifier = getNextPacketIdentifier();
+
+    result = MQTT_SerializeUnsubscribe( mqttSubscription,
+                                        sizeof( mqttSubscription ) / sizeof( MQTTSubscribeInfo_t ),
+                                        unsubscribePacketIdentifier,
+                                        remainingLength,
+                                        pFixedBuffer );
+    assert( result == MQTTSuccess );
+
+    /* Send Unsubscribe request to the broker. */
+    status = send( tcpSocket, ( void * ) pFixedBuffer->pBuffer, packetSize, 0 );
+    assert( status == ( int ) packetSize );
+}
+/*-----------------------------------------------------------*/
+
+static void mqttKeepAlive( int tcpSocket,
+                           MQTTFixedBuffer_t * pFixedBuffer )
+{
+    MQTTStatus_t result;
+    int status;
+    size_t packetSize = 0;
+
+    status = MQTT_GetPingreqPacketSize( &packetSize );
+
+    /* PingReq is fixed length packet, therefore there is no need to calculate the size,
+     * just makes sure static buffer can accommodate ping request. */
+    assert( packetSize <= SHARED_BUFFER_SIZE );
+
+    result = MQTT_SerializePingreq( pFixedBuffer );
+    assert( result == MQTTSuccess );
+
+    /* Send Ping Request to the broker. */
+    status = send( tcpSocket, ( void * ) pFixedBuffer->pBuffer, packetSize, 0 );
+    assert( status == ( int ) packetSize );
+}
+
+/*-----------------------------------------------------------*/
+
+static void mqttDisconnect( int tcpSocket,
+                            MQTTFixedBuffer_t * pFixedBuffer )
+{
+    MQTTStatus_t result;
+    int32_t status;
+    size_t packetSize = 0;
+
+    status = MQTT_GetDisconnectPacketSize( &packetSize );
+
+    assert( packetSize <= pFixedBuffer->size );
+
+    result = MQTT_SerializeDisconnect( pFixedBuffer );
+    assert( result == MQTTSuccess );
+
+    /* Send disconnect packet to the broker */
+    status = send( tcpSocket, ( void * ) pFixedBuffer->pBuffer, packetSize, 0 );
+    assert( status == ( int ) packetSize );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Process a response or ack to an MQTT request (PING, SUBSCRIBE
+ * or UNSUBSCRIBE). This function processes PING_RESP, SUB_ACK, UNSUB_ACK
+ *
+ * @param[in] pxIncomingPacket is a pointer to structure containing deserialized
+ * MQTT response.
+ *
+ * @param[in] packetId is packet identifier from the incoming if it was received.
+ * @note Not all responses contain packet identifier.
+ */
+static void mqttProcessResponse( MQTTPacketInfo_t * pIncomingPacket,
+                                 uint16_t packetId )
+{
+    switch( pIncomingPacket->type & 0xf0 )
```

Comment:
This isn't strictly necessary, since the lower four bits of all of these packets must be 0.

Reply:
True, but I'll leave it there just to emphasize.